### DISCUSSION TITLE
How to override paint() method from subclass of frame?

### DISCUSSION THREAD

**Discussion by jingkaimori** [scala/scala-swing#171] [Q&A]
[docs/SIP-8.md](https://github.com/scala/scala-swing/blob/work/docs/SIP-8.md) describes how to override paint method in subclass of Component, but this method cannot apply to subclass of Frame.

**Comment by Sciss** [comment]
Is it important that it is the frame itself? I think this is a rather low level scenario that was probably not covered for Scala-Swing. You could always install a single component that does the drawing:

```scala
import scala.swing._
import java.awt.Color

new Frame {
  contents = new Component {
    preferredSize = new Dimension(400, 400)
    
    override def paintComponent(g: Graphics2D): Unit = {
      g.setColor(Color.red)
      g.drawLine(0, 0, peer.getWidth, peer.getHeight)
    }
  }
  pack().centerOnScreen()
  open()
}
```
would this be suitable in your case?

**Comment by SethTisue** [comment]
Note that (not just in Scala Swing, but in Swing generally) every `JFrame` has a "content pane"; doing things with a `JFrame` often involves interacting with the content pane, rather than with the frame itself.
